Make sure crypto summary is displayed properly and use `cryptoAmount` instead of `amount`

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/6407

[btc.webm](https://user-images.githubusercontent.com/12435965/217895909-5e7942b1-db85-4029-b6af-9f70d5a84983.webm)

znarf commented:
Have you seen https://github.com/opencollective/opencollective/issues/6401 ?

Can you explain what's happening in details ?

The rationale behind the fix ?

Since when is it broken ? What's the regression ?

znarf commented:
Is it covered by E2E tests ?

znarf commented on components/contribution-flow/ContributionFlowStepsProgress.js:
```diff
@@ -28,8 +28,9 @@ const StepLabel = styled(Span)`
-    const totalAmount = amount + (stepDetails.platformTip || 0);
+    const totalAmount = isCrypto
+      ? stepDetails.cryptoAmount || 0
+      : stepDetails.amount + (stepDetails.platformTip || 0) || 0;
```
Difficult to read this.

SudharakaP replied:
I've used `lodash` to simplify a bit. Should be better now.

znarf commented on components/contribution-flow/ContributionSummary.js:
```diff
@@ -64,9 +64,9 @@ const ContributionSummary = ({
+  const totalAmount = getTotalAmount(stepDetails, stepSummary, isCrypto);
   const pmFeeInfo = getPaymentMethodFees(stepPayment?.paymentMethod, totalAmount, currency);
-  const platformTip = stepDetails.platformTip || 0;
+  const platformTip = !isCrypto ? stepDetails.platformTip || 0 : 0;
```
Weird to read

SudharakaP replied:
I've used `lodash` to simplify a bit. Let me know.

znarf commented on components/contribution-flow/ContributionSummary.js:
```diff
@@ -116,7 +116,7 @@ const ContributionSummary = ({
             ))}
 
-          {Boolean(platformTip) && (
+          {Boolean(platformTip) && !isCrypto && (
```
Why do this if platformTip is already set to zero?

SudharakaP replied:
Ops, this is a mistake. I've remove this condition.

znarf commented:
![Uploading Screenshot 2023-02-10 at 15.37.39.png…]()

This is also broken

znarf commented:
This is also not fixed:

<img width="637" alt="Screenshot 2023-02-10 at 15 50 43" src="https://user-images.githubusercontent.com/806/218121496-2df4b581-8971-4bdc-aeb9-6b2efb3692d6.png">

SudharakaP commented:
> Have you seen [opencollective/opencollective#6401](https://github.com/opencollective/opencollective/issues/6401) ?
> 
> Can you explain what's happening in details ?
> 
> The rationale behind the fix ?
> 
> Since when is it broken ? What's the regression ?

This is a regression introduced when we added the `cryptoAmount` variable here; https://github.com/opencollective/opencollective-frontend/pull/8394. Note that we are sending the wrong `totalAmount` calculated via the `amount` to the API not the correct `cryptoAmount` as I've corrected in this PR. 

> Is it covered by E2E tests ?

Crypto flow isn't covered by E2E tests. We have an issue for this, https://github.com/opencollective/opencollective/issues/4649. I'll work on this soon maybe since given more recent changes to the contribution flow maybe this is important.

SudharakaP commented:
> <img alt="Screenshot 2023-02-10 at 15 37 39" width="1063" src="https://user-images.githubusercontent.com/806/218121549-98691d89-2aa6-41c8-965b-894f4bdf0053.png">
> 
> This is also broken

Fixed this in the sense the correct currency is shown; 

![image](https://user-images.githubusercontent.com/12435965/218205635-f7b8c904-60e2-4e28-a445-fa245ba43022.png)

However this summary screen is only shown when we refresh the page within the contribution flow. I have opened a new issue to investigate as that's not connected to this issue; https://github.com/opencollective/opencollective/issues/6412

znarf commented on components/contribution-flow/ContributionSummary.js:
```diff
@@ -64,9 +65,10 @@ const ContributionSummary = ({
-  const pmFeeInfo = getPaymentMethodFees(stepPayment?.paymentMethod, totalAmount, currency);
-  const platformTip = stepDetails.platformTip || 0;
+  const totalAmount = getTotalAmount(stepDetails, stepSummary, isCrypto);
+  const displayCurrency = isCrypto ? stepDetails.currency.value : currency;
```
Why can't we rely on currency?

znarf commented on components/contribution-flow/StepSummary.js:
```diff
@@ -245,7 +245,7 @@ const StepSummary = ({
   const hostCountry = get(collective.host, 'location.country');
   const collectiveCountry = collective.location.country || get(collective.parent, 'location.country');
-  const currency = tier?.amount.currency || collective.currency;
+  const currency = isCrypto ? stepDetails.currency.value : tier?.amount.currency || collective.currency;
```
linter feedback

znarf replied:
```suggestion
  const currency = isCrypto ? stepDetails.currency.value : (tier?.amount.currency || collective.currency);
```

Easier to read with parenthesis?

SudharakaP replied:
Thanks. Corrected that. 👍🏽

SudharakaP replied:
Done. 👍🏽

SudharakaP replied:
@znarf : Actually this one isn't possible as prettier seem to think that not having parenthesis is better. So either we have to make a prettier exception here or just leave it without parenthesis.
